src/test/java/com/thebois/models/beings/BeingTests.java

```diff
@@ -160,4 +183,79 @@ public void equalsReturnsFalseForNotEqualBeings(final AbstractBeing first,
         assertThat(first).isNotEqualTo(second);
     }
 
+    @Test
+    public void pawnBuildsHouses() {
+        // Arrange
+
+        final IFinder mockFinder = Mockito.mock(IFinder.class);
+        final Position position = new Position();
+        final IStructure structure = new House(position);
+        when(mockFinder.findNearestStructure(any())).thenReturn(Optional.of(structure));
+        final Pawn pawn = new Pawn(position, position, Mockito.mock(Random.class), mockFinder);
+
+        // Act
+        final float initialBuildStatus = structure.builtStatus();
+        pawn.update();
+        final float finalBuildStatus = structure.builtStatus();
+
+        // Assert
+
+        assertThat(initialBuildStatus).isLessThan(finalBuildStatus);
+    }
+
+    @Test
+    public void pawnDoNotBuildsHouses() {
+        // Arrange
+        final IFinder mockFinder = Mockito.mock(IFinder.class);
+        final Position position = new Position(0, 0);
+        final Position farAway = new Position(50, 50);
+        final IStructure structure = new House(farAway);
+        when(mockFinder.findNearestStructure(any())).thenReturn(Optional.of(structure));
+        final Pawn pawn = new Pawn(position, position, Mockito.mock(Random.class), mockFinder);
+
+        // Act
+        final float initialBuildStatus = structure.builtStatus();
+        pawn.update();
+        final float finalBuildStatus = structure.builtStatus();
+
+        // Assert
+        // Value did not change because Pawn was not close enough
+        assertThat(initialBuildStatus).isEqualTo(finalBuildStatus);
+    }
+
+    @Test
+    public void doesPawnWalkToNearestUnBuiltBuilding() {
+        // Arrange
+        final Position positionA = new Position(1, 0);
+        final Position positionB = new Position(2, 0);
+        final Position positionC = new Position(3, 0);
```

Comment:
These names explain nothing about what they represent. 

Is `positionA` the starting position of one of the structures, or is it where the being is located? I don't know because the names are meaningless.